ext/mbstring/libmbfl/filters/mbfilter_utf32.c

```diff
@@ -289,3 +311,107 @@ int mbfl_filt_conv_wchar_utf32le(int c, mbfl_convert_filter *filter)
 
 	return c;
 }
+
+static int mbfl_filt_ident_utf32(int c, mbfl_identify_filter *filter)
+{
+	/* The largest valid codepoint is 0x10FFFF; all we care about is that
+	 * we don't find a value bigger than that
+	 * For UTF-32 (not LE or BE), we do also need to look for a byte-order mark */
```

Comment:
Per the answer you quoted and Wikipedia:

> In November 2003, Unicode was restricted by RFC 3629 to match the constraints of the UTF-16 encoding: explicitly prohibiting code points greater than U+10FFFF (and also the high and low surrogates U+D800 through U+DFFF). This limited subset defines UTF-32.

So, do we need to also check for surrogates? Though later it does say:

> Though technically invalid, the surrogate halves are often encoded and allowed.

Do you know whether we allow surrogates in our UTF-8 implementation?

Reply:
No, mbstring handles surrogates correctly for UTF-8 (as well as UTF-32, etc). :smile:

I am just writing some rigorous testing code for UTF-{7,8,16,32} to make sure that there are no lurking bugs for these encodings. In the meantime, if you have time, I would encourage you to look at some of the commits here which improve the handling of ISO-8859-x, SJIS-2004, SJIS-mac, and so on. Maybe some of those can be landed.

Reply:
> No, mbstring handles surrogates correctly for UTF-8 (as well as UTF-32, etc).

Oh yes, I'm not sure how I could have missed that in the UTF-32 implementation, it's quite obvious there... In that case we should be rejecting them in the identify filter as well though.